gobblin-runtime/src/main/java/org/apache/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -463,6 +469,20 @@ public void apply(JobListener jobListener, JobContext jobContext)
           return;
         }
 
+        // calculation of total bytes to copy in a job used to track a job's copy progress
+        if (jobState.getPropAsBoolean(ConfigurationKeys.REPORT_JOB_PROGRESS, false)) {
+          if (workUnitStream.isSafeToMaterialize()) {
+            Collection<WorkUnit> workUnits = JobLauncherUtils.flattenWorkUnits(workUnitStream.getMaterializedWorkUnitCollection());
+            long totalSizeInBytes = sumWorkUnitsSizes(workUnits);
+            this.jobContext.getJobState().setProp(TOTAL_BYTES_TO_COPY, totalSizeInBytes);
+          } else {
+            throw new RuntimeException("Property " + ConfigurationKeys.REPORT_JOB_PROGRESS + " is turned on, but "
+                + "progress cannot be collected because " + DestinationDatasetHandlerService.class.getName()
+                + " does not support work unit streams. Turn off property " + ConfigurationKeys.REPORT_JOB_PROGRESS
+                + " and rerun job.");
```

Comment:
I think it's not necessarily true that it's due to `DestinationDatasetHandler` not supporting workunitstreams, but that this feature also does not support work unit streams (as total size is determined only once at the beginning).

Reply:
We can put a better exception message here to let the user know that copy progression doesn't work for infinite streams (which make sense, since how would we determine that the work is completed?)